src/building/bootstrapping.md

```diff
@@ -47,54 +62,59 @@ graph TD
     classDef with-s1c fill: lightgreen;
 ```
 
-### Stage 0
+[objects]: https://en.wikipedia.org/wiki/Object_code
+
+### The stages: how each compiler is obtained
+
+#### Stage 0: the pre-compiled compiler
 
-The stage0 compiler is usually the current _beta_ `rustc` compiler
-and its associated dynamic libraries,
-which `x.py` will download for you.
-(You can also configure `x.py` to use something else.)
+A pre-compiled compiler and its set of dependencies are downloaded. By default,
+it is the current beta release. This is the stage 0 compiler.
 
-The stage0 compiler is then used only to compile `rustbuild`, `std`, and `rustc`.
-When compiling `rustc`, the stage0 compiler uses the freshly compiled `std`.
-There are two concepts at play here:
-a compiler (with its set of dependencies)
-and its 'target' or 'object' libraries (`std` and `rustc`).
-Both are staged, but in a staggered manner.
+#### Stage 1: from current code, by an earlier compiler
 
-### Stage 1
+The stage 0 compiler compiles from current code `rustbuild` and `std` and uses
+them to compile from current code a compiler. This is the stage 1 compiler.
 
-The rustc source code is then compiled with the stage0 compiler to produce the stage1 compiler.
+The stage 1 compiler is the first that is from current code. Yet, it is not
+entirely up-to-date, because the compiler that compiled it is of earlier code.
+More on this below.
 
-### Stage 2
+#### Stage 2: the truly current compiler
 
-We then rebuild our stage1 compiler with itself to produce the stage2 compiler.
+By default, the stage 1 libraries are copied into stage 2, because they are
+expected to be identical.
 
-In theory, the stage1 compiler is functionally identical to the stage2 compiler,
-but in practice there are subtle differences.
-In particular, the stage1 compiler itself was built by stage0
-and hence not by the source in your working directory.
-This means that the symbol names used in the compiler source
-may not match the symbol names that would have been made by the stage1 compiler,
-which can cause problems for dynamic libraries and tests.
+The stage 1 compiler is used to compile from current code a compiler. This is
```

Comment:
*produce* (or equivalent) also missing here

Reply:
Resolve in favor of https://github.com/rust-lang/rustc-dev-guide/pull/1458#discussion_r1048000762 ?